File: aibom/src/aibom/models/component.py
# Copyright 2026 Cisco Systems, Inc. and its affiliates
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
# SPDX-License-Identifier: Apache-2.0

from typing import Optional, Dict, Any
from pydantic import BaseModel


class ComponentModel(BaseModel):
    """Pydantic model for AI components."""
    id: Optional[str] = None
    name: str
    file_path: str
    line_number: int
    type: str  # category/type of component
    text: Optional[str] = None
    model_name: Optional[str] = None
    embedding_model: Optional[str] = None
    additional_data: Optional[Dict[str, Any]] = None


class ComponentResponse(BaseModel):
    """Response model for component API."""
    components: list[ComponentModel]
    total: int


class ComponentTypesResponse(BaseModel):
    """Response model for component types API."""
    types: list[str]

File: ui/src/types/component.ts
// Copyright 2026 Cisco Systems, Inc. and its affiliates
//
// SPDX-License-Identifier: Apache-2.0

export interface Component {
  id: string;
  name: string;
  file_path: string;
  line_number: number;
  type: 'prompt' | 'retriever' | 'datastore' | 'tool' | 'memory' | 'agent' | 'other' | string;
  text: string | null;
  model_name: string | null;
  embedding_model: string | null;
  additional_data?: Record<string, unknown> | null;
}
